lib/storage/storage.go

```diff
@@ -251,6 +255,16 @@ func MustOpenStorage(path string, opts OpenOptions) *Storage {
 	s.pendingNextDayMetricIDs = &uint64set.Set{}
 
 	s.prefetchedMetricIDs = &uint64set.Set{}
+	if opts.TrackMetricNamesStats {
+		mnt := metricnamestats.MustLoadFrom(filepath.Join(s.cachePath, "metric_usage_tracker"), uint64(getMetricNamesStatsCacheSize()))
+		s.metricsTracker = mnt
+		if mnt.IsEmpty() {
```

Comment:
this should be checked before `s.metricsTracker = mnt`. Otherwise, tracker could be updated before Empty check.
Btw, locally I wasn't able to get TSID cache reseted automatically here. I assume this concurrent access is the problem, but it could be something else.